modules/snooker/src/main/ts/ephox/snooker/api/CellMutations.ts

```diff
@@ -1,13 +1,20 @@
 import { SugarElement } from '@ephox/sugar';
 import * as Sizes from '../resize/Sizes';
+import * as CellUtils from '../util/CellUtils';
 
 const halve = (main: SugarElement, other: SugarElement): void => {
-  const width = Sizes.getGenericWidth(main);
-  width.each((w) => {
-    const newWidth = w.value / 2;
-    Sizes.setGenericWidth(main, newWidth, w.unit);
-    Sizes.setGenericWidth(other, newWidth, w.unit);
-  });
+  const colspan = CellUtils.getSpan(main, 'colspan');
+  // Don't set a width on the new cell if we have a colspan as we can only safely do that for cells
+  // that are a single column, since we don't know the individual column widths for a cell with a colspan.
+  // Instead, we'll rely on the adjustments/postAction logic to set the widths based on other cells in the column
+  if (colspan === 1) {
```

Comment:
Can we use `hasColspan` here?

```suggestion
  if (!hasColspan(main)) {
```

Reply:
You could have `colspan="1"` in the HTML, so no that wouldn't work.

Reply:
Oh geez, the implementation does handle that, but it's not clear from the names so I definitely don't like that. I also doesn't handle the case where someone might have `colspan="0"` or something invalid.